src/component/registry/bpmn-model-registry.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2021 Bonitasoft S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import BpmnModel from '../../model/bpmn/internal/BpmnModel';
+import Shape from '../../model/bpmn/internal/shape/Shape';
+import Edge from '../../model/bpmn/internal/edge/Edge';
+import { BpmnSemantic } from './types';
+import ShapeUtil from '../../model/bpmn/internal/shape/ShapeUtil';
+import ShapeBpmnElement, { ShapeBpmnSubProcess } from '../../model/bpmn/internal/shape/ShapeBpmnElement';
+import { ShapeBpmnMarkerKind } from '../../model/bpmn/internal/shape';
+
+export class BpmnModelRegistry {
+  private searchableModel: SearchableModel;
+
+  computeRenderedModel(bpmnModel: BpmnModel): RenderedModel {
+    this.searchableModel = new SearchableModel(bpmnModel);
+    return toRenderedModel(bpmnModel);
+  }
+
+  getBpmnSemantic(bpmnElementId: string): BpmnSemantic | undefined {
+    const element = this.searchableModel.elementById(bpmnElementId);
+    if (!element) {
+      return undefined;
+    }
+    const bpmnElement = element.bpmnElement;
+    const isShape = bpmnElement instanceof ShapeBpmnElement;
+    return { id: bpmnElementId, name: bpmnElement.name, isShape: isShape, kind: bpmnElement.kind };
+  }
+}
+
+function toRenderedModel(bpmnModel: BpmnModel): RenderedModel {
```

Comment:
⚠️ Sonar suggests that this function is not fully covered by unit and integration tests
We should at least model tests for all use cases.

![image](https://user-images.githubusercontent.com/27200110/109468481-ec08fa00-7a6c-11eb-8b1f-82ae64022c25.png)

Reply:
This is managed by the removal of the 'safe navigation operator' `?.` in 7dd96bf